specification/storageimportexport/resource-manager/Microsoft.ImportExport/2016-11-01/storageimportexport.json

```diff
@@ -1,1210 +1,1098 @@
+                            "items":{
+                                "type": "string"
+                            }
+                        },
+                        "alternateLocations":{
+                            "type": "array",
+                            "description": "A list of location IDs that should be used to ship shipping drives to for jobs created against the current location. If the current location is active, it will be part of the list. If it is temporarily closed due to maintenance, this list may contain other locations. ",
+                            "items":{
+                                "type": "string"
+                            }
+                        }
+                    }
+                }
+            }
         },
-        "cancelRequested": {
-          "type": "boolean",
-          "description": "Indicates whether a request has been submitted to cancel the job."
+        "ReturnAddress": {
+            "description": "Specifies the return address information for the job.",
+            "properties": {
+                "recipientName": {
+                    "type": "string",
+                    "description": "The name of the recipient who will receive the hard drives when they are returned. "
+                },
+                "streetAddress1": {
+                    "type": "string",
+                    "description": "The first line of the street address to use when returning the drives. "
+                },
+                "streetAddress2": {
+                    "type": "string",
+                    "description": "The second line of the street address to use when returning the drives. "
+                },
+                "city": {
+                    "type": "string",
+                    "description": "The city name to use when returning the drives."
+                },
+                "stateOrProvince": {
+                    "type": "string",
+                    "description": "The state or province to use when returning the drives."
+                },
+                "postalCode": {
+                    "type": "string",
+                    "description": "The postal code to use when returning the drives."
+                },
+                "countryOrRegion": {
+                    "type": "string",
+                    "description": "The country or region to use when returning the drives. "
+                },
+                "phone": {
+                    "type": "string",
+                    "description": "Phone number of the recipient of the returned drives."
+                },
+                "email": {
+                    "type": "string",
+                    "description": "Email address of the recipient of the returned drives."
+                }
+            }
         },
-        "percentComplete": {
-          "readOnly": true,
-          "type": "integer",
-          "description": "Overall
```

Comment:
your PR removes `required` from the following type definitions: `Operation`, `ReturnAddress`, `ReturnShipping`, `ShippingInformation`, `PackageInfomation`, `DriveStatus` and possibly more (I think it's easiest if you could just do a quick pass over all types and see if something is required - looks as if the removal was by mistake, since the "old" version of the spec had them, the "new" one does not)

Reply:
Operation type can only be used in ListOperation response. Making some properties to be required does not provide any information. Removing it makes much more sense.

Reply:
Restored other required fields: ReturnAddress, ReturnShipping, ShippingInformation, PackageInfomation, DriveStatus